### DISCUSSION TITLE
Best way to update the docker images

### DISCUSSION THREAD

**Discussion by faethon** [ggogel/seafile-containerized#216] [Q&A]
So, I found this containerized version of seafile. It is a much more transparent and well configured docker way of deploying seafile. Looking at the docker-compose file I noticed specific version tags for all images. This makes sense, since it is known stable config. But as time progresses it will likely be updated. I could go into my adapted docker-compose file and update accordingly as new versions are posted here. However, a more lean and mean way would be helpful, where I do not have to update the version tags manually when a new release of the docker images is being pushed onto docker hub.

So I looked at the different versions hardcoded in the docker-compose.yml file, and noticed the following:

`ggogel/seafile-server hardcoded` version tag (11.0.7 as of writing) is same as `latest`
`ggogel/seahub hardcoded` version tag (11.0.7 as of writing) is same as `latest`
`ggogel/seahub-media` version tag (11.0.7 as of writing) is same as `latest`
`mariadb` version tag (10.11.7 as of writing) is same as `lts`
`memcached` version tag (1.6.26 as of writing) is same as `latest`
`ggogel/seafile-caddy` version tag (2.7.6 as of writing) is same as `latest`

So I assume replacing the version tags with the corresponding label tags, would keep things in line with the latest release. Then, when a new version is pushed by ggogel, just rerunning docker compose would pull the latest versions out. Obviously this may introduce a risk of getting errors, but done manually this should be easy to overcome I guess. Is this a valid update strategy, even in a somewhat production environment?

I am wondering what the best update strategy would be. And would like to discuss what would be recommended.

**Comment by sahilph** [comment]
I would just manually change the version tags then:
```
docker compose pull
docker compose up -d
```
Using latest tags is not recommended as it may have some breaking changes.

**Comment by faethon** [comment]
For now I have used the `latest` and `lts` tags and will do a pull when a new version is released. I have confidence this will work out alright, but I agree manually updating the tags will give full control. I'll see what happens in the coming months if when a new releases will be posted.

I am curious though, but assume that @ggogel verifies if a new release works before pushing the `latest` tag. I assume so.

**Comment by sahilph** [comment]
> I am curious though, but assume that @ggogel verifies if a new release works before pushing the `latest` tag. I assume so.

Yes, most likely he does verify it. But sometimes mistakes happen, so I feel its better to be safe than to be sorry.  😉
